qiskit/transpiler/passes/dynamical_decoupling/xy4.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""XY4 is a basic dynamical decoupling (DD) sequence that applies corrective pulses around
+both the X and Y axes, to correct multiple components of the system-environment interaction.
+The sequence is comprised of an X gate, Y gate, X gate, and Y gate with a fixed delay between
+each pulse.
+
+This implementation saturates idle periods of any qubit with XY4 sequences.
+"""
+from qiskit.circuit.delay import Delay
+from qiskit.circuit.library.standard_gates import XGate, YGate
+from qiskit.circuit.quantumregister import QuantumRegister, Qubit
+from qiskit.dagcircuit import DAGCircuit
+
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.passes.basis.unroller import Unroller
+
+
+class XY4(TransformationPass):
+    """Pass which inserts XY4 sequences into a scheduled circuit where there are delays
+    of sufficent duration.
+    """
+
+    def __init__(self, backend_properties, tau_step=10e-9):
+        """XY4 pass initializer.
+
+        Args:
+            backend_properties (BackendProperties): Properties returned by a
+                backend, including information on gate errors, readout errors,
+                qubit coherence times, etc.
+            tau_step (float): Delay time between pulses in the DD sequence in
+                seconds. Default is 10 ns.
+        """
+        super().__init__()
+        self.backend_properties = backend_properties
+        self.tau_step = tau_step
+        """To do"""
+        self.tau_cs = {}
+
+        basis = backend_properties.gate_names()  # TODO
+        num_qubits = len(backend_properties.qubits)
+
+        for qubit in range(num_qubits):
```

Comment:
Could the code in this block be broken out into some private class methods with a brief comment for what is happening? It seem like the purpose is to calculate the duration of an XY4 sequence for each qubit?